packages/amplication-server/src/core/git/git.provider.service.ts

```diff
@@ -326,6 +335,80 @@ export class GitProviderService {
     return await gitClientService.getGitInstallationUrl(workspaceId);
   }
 
+  async getCurrentOAuthUser(oAuthUserName: string): Promise<GitOrganization> {
+    return this.prisma.gitOrganization.findFirst({
+      where: { name: oAuthUserName },
+    });
+  }
+
+  async completeOAuth2Flow(
+    args: CompleteGitOAuth2FlowArgs
+  ): Promise<GitOrganization> {
+    const { code, gitProvider, workspaceId } = args.data;
+    const gitClientService = await new GitClientService().create(
+      {
+        provider: gitProvider,
+        installationId: null,
+      },
+      this.logger
+    );
+    try {
+      const {
+        accessToken,
+        refreshToken,
+        expiresIn,
+        tokenType,
+        scopes,
+        userData: { name, uuid },
+      } = await gitClientService.completeOAuth2Flow(code);
+
+      return this.prisma.gitOrganization.upsert({
+        where: {
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          provider_installationId: {
+            provider: gitProvider,
+            installationId: uuid,
+          },
+        },
+        create: {
+          provider: gitProvider,
+          installationId: uuid,
+          name,
+          type: EnumGitOrganizationType.User,
+          workspace: {
+            connect: {
+              id: workspaceId,
+            },
+          },
+          providerProperties: {
+            username: name,
+            uuid,
+            accessToken,
+            refreshToken,
+            expiresIn,
+            tokenType,
+            scopes,
+          },
+        },
+        update: {
+          name: name,
+          providerProperties: {
+            username: name,
+            uuid,
+            accessToken,
+            refreshToken,
+            expiresIn,
+            tokenType,
+            scopes,
+          },
+        },
+      });
+    } catch (error) {
+      console.error(error);
```

Comment:
@abrl91 we need to use the new logger, also, why throw? this is what we want to do?